src/v/context/context_frame.h

```diff
@@ -0,0 +1,253 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+/// \file context_frame.h
+/// \brief Mixin-composable context frame.
+///
+/// This header provides `context_frame<Mixins...>` for composing contexts with
+/// zero-cost, opt-in features. Most users only need `context/context.h` for
+/// `context_ref` and `background()`.
+///
+/// ## Usage
+///
+/// ```cpp
+/// #include "context/context_frame.h"
+/// #include "context/deadline_timer.h"
+///
+/// using my_frame = context::context_frame<context::deadline_timer>;
+/// my_frame frame{parent, context::with<context::deadline_timer>(100ms)};
+/// ```
+///
+/// ## Built-in Mixins
+///
+/// - `deadline_timer` (deadline_timer.h): Automatic cancellation on timeout.
+/// - `abort_source` (abort_source.h): Bridge to Seastar's abort_source.
```

Comment:
I think I would be happier if we `s/context_frame/context` because I really see all these as "contexts" and you get references to them. Because "context_ref" is really a reference to a frame not a context, so it should be called `context_frame_ref`? I think having a simple `context` and `context_ref` as the two main classes is simpler, but I have to learn this mixin thing more first.

My gut reaction is the mixin thing feels like it's a bit over complicated, but I can probably be convinced.